build.assets/tooling/cmd/resource-ref-generator/README.md

```diff
@@ -0,0 +1,81 @@
+# Resource reference generator
+
+The resource reference generator is a Go program that produces a comprehensive
+reference guide for all dynamic Teleport resources and the fields they include.
+It uses the Teleport source as the basis for the guide. 
+
+## Usage
+
+```
+$ cd build.assets/tooling/cmd/resource-ref-generator
+$ go run . -config config.yaml
```

Comment:
Consider adding a make target to the root makefile.

Reply:
So far, the root Makefile has targets for checking that auto-generated docs are up to date ([here](https://github.com/gravitational/teleport/blob/d15466c5371fd8a2456932891afbffe0b2a67cb6/Makefile#L1606-L1628)), but not for generating the docs. I could follow this convention and put the target in [`build.assets/Makefile`](https://github.com/gravitational/teleport/blob/master/build.assets/Makefile), though a comment in the file says this is for producing official Teleport releases. At the same time, a root make target for running the docs generators would be nice, since they're spread throughout the source.

Come to think of it, is `build.assets/tooling/cmd` the best place for this project? Would it make sense to place it in `docs`?

Reply:
Added